### ISSUE TITLE
Highlighter should interpret Helm directives before YAML lists

### ISSUE THREAD

**Issue by jcwilkes** [Azure/vscode-kubernetes-tools#805]
When highlighting the following snippet

```yaml
args: [ {{ join "," .Values.args }} ]
```
 the comma is interpreted as being a YAML list separator instead of part of the quoted string in the join directive. This causes the second quotation mark after the comma to be interpreted as the start of the new string and causes bad highlighting for the rest of the file.

---

**Kubernetes Extension** v1.2.1

**VSCode**
```
Version: 1.47.3 (user setup)
Commit: 91899dcef7b8110878ea59626991a18c8a6a1b3e
Date: 2020-07-23T13:12:49.994Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.19041
```

**Comment by itowlson** [comment]
This may be tricky - we lean on the Red Hat YAML parser for Helm charts, but I think Helm charts are not actually YAML.  I'm not sure how to address this short of forking the YAML parser and creating a variant that understands Go template syntax.  @bacongobbler as the resident Helm whiz, do you have any ideas on ways we could tackle this?  Even if they're kludgy?

**Comment by bacongobbler** [comment]
I agree with your point @itowlson. The correct path forward would be to write a `text/template` highlighter. I have not yet heard of one at this time however.

Some discussion on vscode-go maintaining a syntax highlighter was declined by the maintainers, but I'm willing to bet someone has attempted to write one. https://github.com/microsoft/vscode-go/issues/464#issuecomment-580093228
